opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -605,22 +605,25 @@ func resourceOpennebulaVirtualMachineRead(ctx context.Context, d *schema.Resourc
 
 		var diags diag.Diagnostics
 
-		// read template ID from which the VM was created
-		templateID, _ := vmInfos.Template.GetInt("TEMPLATE_ID")
-		d.Set("template_id", templateID)
-
-		// add empty values for import
-		if _, ok := d.GetOk("template_disk"); !ok {
-			d.Set("template_disk", []interface{}{})
-		}
-		if _, ok := d.GetOk("template_nic"); !ok {
-			d.Set("template_nic", []interface{}{})
-		}
-		if _, ok := d.GetOk("template_tags"); !ok {
-			d.Set("template_tags", map[string]interface{}{})
-		}
-		if _, ok := d.GetOk("template_section_names"); !ok {
-			d.Set("template_section_names", map[string]interface{}{})
+		// NOTE: The template_* attributes are not defined for Virtual Router instances (VMs).
+		if _, ok := d.GetOk("template_id"); ok {
```

Comment:
why adding this check ?
It's possible to inherit disks and tags from virtual router instance template, right ?
However, maybe that `template_nic` code is not relevant here.

The schema description (`template_nic` etc.) is added via this func: https://github.com/OpenNebula/terraform-provider-opennebula/blob/master/opennebula/resource_opennebula_virtual_router_instance.go#L41

Reply:
I'm not completely sure if this implementation is correct either, hence the review :smiling_face_with_tear:. But, if you look at the `opennebula_virtual_router_instance` resource, there is no `template_id` argument :thinking:. At the same time VR instances are processed with code reused from handling VMs (that have optional `template_id`  argument), so the `d.Set("template_id", templateID)`  basically causes panic in this specific case context is updated. Perhaps I should rephrase the comment as I confused attributes with arguments. :thinking:

Reply:
@treywelsh Hi, I've been MIA sorry. So indeed attributes defined here https://github.com/OpenNebula/terraform-provider-opennebula/blob/master/opennebula/shared_schemas.go#L20 can be kept and they don't cause panics, `template_id` and `template_nic` on the other hand do. :face_holding_back_tears: